### PR TITLE
fix: remove alert channel properly if test fails

### PR DESCRIPTION
**Pull request by afiune**
***Issue***:
* https://lacework.atlassian.net/browse/ALLY-688
* https://github.com/lacework/terraform-provider-lacework/issues/181

***Description:***
When users create new alert channels, by default we test the
integrations, when it fails, we remove it from their Lacework account
but we forgot to remove it from the Terraform state.

We are fixing
https://github.com/lacework/terraform-provider-lacework/issues/181 by
removing the alert channel from the Terraform state.

Signed-off-by: Salim Afiune Maya <afiune@lacework.net>

***Additional Info:***
To test this, go into any example in our `examples/` folder, switch the flag `test_integration`
to `true` and run Terraform. On a second `terraform apply` you should not see the message:
```
│ Error:
│   [POST] https://account.lacework.net/api/v2/AlertChannels/ABC_500FC239AFE4CAF64C4704382EC43111D01A57AC16E94B7/test
│   [404] no_team
```

### CODE REVIEW

**Review by dmurray-lacework — APPROVED** [review]
1 typo to fix

File: integration/integration.go [hunk 1]
```diff
@@ -165,3 +155,12 @@ func GetAlertChannelProps(result string) api.AlertChannelResponse {
 	}
 	return data
 }
+
+func GetIDFromTerraResults(result string) string {
```
Comment (vatasha) [inline integration/integration.go:159]: Nice!

File: lacework/alert_channel_importer.go [hunk 2]
```diff
@@ -12,10 +12,11 @@ import (
...
 // if the test is not successful, it will remove the alert channel (rollback)
-func VerifyAlertChannelAndRollback(id string, lacework *api.Client) error {
-	if err := lacework.V2.AlertChannels.Test(id); err != nil {
+func VerifyAlertChannelAndRollback(d *schema.ResourceData, lacework *api.Client) error {
+	if err := lacework.V2.AlertChannels.Test(d.Id()); err != nil {
+		defer d.SetId("")
 		// rollback terraform create upon error testing integration
-		if deleteErr := lacework.V2.AlertChannels.Delete(id); deleteErr != nil {
+		if deleteErr := lacework.V2.AlertChannels.Delete(d.Id()); deleteErr != nil {
 			return errors.Wrapf(deleteErr, "Unable to rollback changes: %v", err)
```
Comment (vatasha) [inline lacework/alert_channel_importer.go:20]: Let's use a domain specific error here as presented in the article above.

**Review by vatasha — CHANGES_REQUESTED** [review]
I would like to see the way we do error checking done a little differently if you're open to my suggestions :)

File: lacework/resource_lacework_alert_channel_victorops.go [hunk 3]
```diff
@@ -193,31 +193,4 @@ func resourceLaceworkAlertChannelVictorOpsDelete(d *schema.ResourceData, meta in
 
 	log.Printf("[INFO] Deleted %s integration with guid %s\n", api.VictorOpsAlertChannelType, d.Id())
 	return nil
-}
-
-func validateVictorOpsAlertChannelResponse(response *api.VictorOpsAlertChannelResponse) error {
```
Comment (afiune) [inline lacework/resource_lacework_alert_channel_victorops.go:198]: I think we forgot to remove this code. cc @dmurray-lacework I removed it for you 😉

**Pull request merged**